config.py

```diff
@@ -1,72 +1,77 @@
+
+This modules `setup()` function does not need to be called directly, it will be called
+    automatically upon first import.
 
 Copyright (c) 2018 The Fuel Rats Mischief,
 All rights reserved.
 
 Licensed under the BSD 3-Clause License.
 
-See LICENSE.md
-
-This module is built on top of the Pydle system.
-
+See LICENSE
 """
+import json
 import logging
+import os
+from typing import Union
 
+from Modules import cli_manager  # For CLI config-file argument
+
+config: Union[None, dict] = None
 
-class IRC:
-    """
-    IRC Configuration
-    """
-    ####
-    # Mecha's presence in IRC
-    presence = "Mechasqueak3-unknown[BOT]"
-    ####
-    # Server to connect to
-    server = "dev.localecho.net"
-    ####
-    # Port to connect on
-    port = "6667"
-    ####
-    # use TLS
-    tls = False
-    ####
-    # what channels to connect to
-    channels = ["#unkn0wndev"]
-
-    class Authentication:
-        """
-        Bots Authentication configuration
-
-        Currently this is just a skeleton and has no effect.
-        """
-        # TODO: Implement SASL authentication
-        username = ""
-        password = ""
-        SASL = True
-
-
-class Logging:
-    """
-    Log configurations
-    """
-    ####
-    # Base logger facility, all others are derivatives
-    base_logger = 'mecha'
-    log_file = f"logs/{IRC.presence}.log"
-    verbosity = logging.DEBUG
 
+def setup_logging(root_logger: str, logfile: str):
+    # create a log formatter
+    log_formatter = logging.Formatter("{levelname} [{name}::{funcName}]:{message}",
+                                      style='{')
+    # get Mecha's root logger
+    log = logging.getLogger(root_logger)
+    # Create a file handler for the logger
+    log_file_handler = logging.FileHandler(logfile, 'w')
+    log_file_handler.setFormatter(log_formatter)
+    # create a stream handler ( prints to STDOUT/STDERR )
+    log_stream_handler = logging.StreamHandler()
+    log_stream_handler.setFormatter(log_formatter)
+    # adds the two handlers to the logger so they can do their thing.
+    log.addHandler(log_file_handler)
+    log.addHandler(log_stream_handler)
+    # set the minimum severity the logger will report.
+    # uncomment for production:
+    # log.setLevel(logging.INFO)
+    # uncomment for develop:
+    log.setLevel(logging.DEBUG)
 
-class Commands:
-    ####
-    # Mecha's trigger prefix
-    trigger = "!"
+    logging.info("[Mecha] configuration file loading...")
+    """provides facilities for managing a configuration from disk"""
 
 
-class RatBoard:
+def setup(filename: str) -> None:
     """
-    Configuration options for the RatBoard
+    Sets up the module by loading the specified configuration file from disk
+
+    Args:
+        filename (str): path and filename to load.
     """
-    CASE_LIMIT = 30
-    """Default limit to board indexies, may be dynamically re-sized during runtime."""
+    global config
+
+    path = f"config/{filename}"
```

Comment:
I believe the config path cli argument should not be relative to the config directory as it's not only somewhat against UNIX convention, will confuse users and breaks autocompletion in the console, it also defeats half the reason for there being a cli argument to set the config file: Being able to set any file in my file system as the config file.